c7n/resources/cw.py

```diff
@@ -315,3 +316,74 @@ def process_resource_set(self, client, accounts, resources):
             if found:
                 results.append(r)
         return results
+
+
+@LogGroup.action_registry.register('set-encryption')
+class EncryptLogGroup(BaseAction):
+    """Encrypt/Decrypt a log group
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: encrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: absent
+            actions:
+              - type: set-encryption
+                kms_key: kms:key:arn
+                state: True
+
+          - name: decrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: kms:key:arn
+            actions:
+              - type: set-encryption
+                state: False
+    """
+    schema = type_schema(
+        'set-encryption',
+        kms_key={'type': 'string'},
+        state={'type': 'boolean'})
+    permissions = ('logs:AssociateKmsKey', 'logs:DisassociateKmsKey')
+
+    def validate(self):
+        self.key = self.data.get('kms_key')
+
+        if not self.data.get('state', True):
+            return self
+
+        if not self.key:
+            raise ValueError('Must specify either a KMS key ARN or Alias')
+        return self
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('logs')
+
+        if self.key:
```

Comment:
if you run the policy with validate off, this would never get set and throw a keyerror as self.key is only set in the validate function

PR here: https://github.com/capitalone/cloud-custodian/pull/2208

additionally the entire key resolution here can be pulled out, I know that s3 bucket encryption does a similar check if key is an alias, if so grab the arn from kms so we should look into moving that into a separate function if it isn't already.